src/annotate/onclass/config.vsh.yaml

```diff
@@ -0,0 +1,140 @@
+        example: reference.h5mu
+        direction: input
+        required: false
+      - name: "--reference_layer"
+        type: string
+        description: The layer in the reference data to be used for cell type annotation if .X is not to be used.
+        required: false
+      - name: "--reference_obs_targets"
+        type: string
+        description: #TODO: update for multiple: Key in obs field of reference AnnData with cell-type information.
+        example: ["cell_ontology_class", "celltype"]
+        default: "cell_ontology_class"
+        multiple: true
+  - name: Outputs
+    description: Output arguments.
+    arguments:
+      - name: "--output"
+        type: file
+        description: Output h5mu file.
+        direction: output
+        example: output.h5mu
+      - name: "--output_compression"
+        type: string
+        choices: ["gzip", "lzf"]
+        required: false
+        example: "gzip"
+      - name: "--output_obs_predictions"
+        type: string
+        example: 
+        required: false
+        multiple: true
+        description: |
+          In which `.obs` slots to store the predicted information.
+          If provided, must have the same length as `--reference_obs_targets`.
+          If empty, will default to the `reference_obs_targets` combined with the `"_pred"` suffix.
+      - name: "--output_obs_probability"
+        type: string
+        example: 
+        required: false
+        multiple: true
+        description: |
+          In which `.obs` slots to store the uncertainty of the predictions.
+          If provided, must have the same length as `--reference_obs_targets`.
+          If empty, will default to the `reference_obs_targets` combined with the `"_probability"` suffix.
+  - name: Model arguments
+    description: Model arguments.
+    arguments:
+      - name: "--model"
+        type: string
+        description: | 
+          "Pretrained model path without a file extension. If not provided, the model will be trained 
+          on the reference data and --reference should be provided. The path namespace should contain:
+            - a .npz or .pkl file
+            - a .data file
+            - a .meta file
+            - a .index file
+          e.g. /path/to/model/pretrained_model_target1 as saved by OnClass."
+        required: false
+        multiple: true
+        example: [pretrained_model_target1; pretrained_model_target2]
+        direction: input
+      - name: "--max_iter"
+        type: integer
+        default: 100
+        required: false
+        description: Maximum number of iterations for training the model.
+
+resources:
+  - type: python_script
+    path: script.py
+  - path: /src/utils/setup_logger.py
+
+test_resources:
+  - type: python_script
+    path: test.py
+  - path: /resources_test/annotation_test_data/
+  - path: /resources_test/pbmc_1k_protein_v3/
+
+engines:
+  - type: docker
+    image: python:3.8
```

Comment:
Can it be updated to a later version? Is slim version also possible?

Reply:
not to a later one because of onclass dependencies. the slim version misses quite a few apt packages so i think it's better to keep the full one